projects/igniteui-angular/src/lib/navigation-drawer/themes/_base.scss

```diff
@@ -0,0 +1,265 @@
+            transform: translate3d(-300px, 0, 0);
+
+            [dir='rtl'] & {
+                transform: translate3d(300px, 0, 0);
+            }
+
+            box-shadow: none;
+
+            &.igx-nav-drawer__aside--right {
+                transform: translate3d(300px, 0, 0);
+                box-shadow: none;
+
+                [dir='rtl'] & {
+                    transform: translate3d(-300px, 0, 0);
+                }
+            }
+        }
+
+        @include e(aside, right) {
+            inset-inline-start: auto;
+            inset-inline-end: 0;
+            border-inline-end: none;
+            border-inline-start: rem(1px) solid var-get($_theme, 'border-color');
+        }
+
+        @include e(aside, mini) {
+            transition-duration: .3s;
+            width: #{var(--igx-nav-drawer-size--mini)};
+            min-width: fit-content;
+
+            %item {
+                justify-content: center;
+                min-width: fit-content;
+
+                igx-icon {
+                    margin-inline-start: 0;
+                }
+            }
+        }
+
+        @include e(aside, normal) {
+            transition-duration: .3s;
+            width: var(--igx-nav-drawer-size);
+        }
+
+        @include e(aside, pinned) {
+            position: relative;
+            box-shadow: none;
+            z-index: 0;
+
+            &.igx-nav-drawer__aside--collapsed {
+                transform: none;
+                width: 0;
+                overflow: hidden;
+                border: none;
+                padding: 0;
+            }
+        }
+
+        @include e(overlay) {
+            opacity: 1;
+            background: color(null, 'gray', 500, .54);
+            transition: opacity, visibility;
+            transition-duration: .3s, .3s;
+            transition-timing-function: ease-in, step-start;
+            transition-delay: 0s, 0s;
+            position: absolute;
+            inset-inline-start: 0;
+            top: 0;
+            width: 100%;
+            height: 100%;
+            visibility: visible;
+            z-index: 999;
+
+            &.panning {
+                transform: translate3d(0, 0, 0);
+                transition: none;
+
+                &.igx-nav-drawer__overlay--hidden {
+                    /* must be visible during pan.. */
+                    visibility: visible;
+                }
+            }
+        }
+
+        @include e(overlay, hidden) {
+            transition-timing-function: ease-in-out, step-end;
+            visibility: hidden;
+            opacity: 0;
+        }
+
+        // Style Dummy(hidden) used for measures
+        @include e(style-dummy) {
+            height: 0;
+            background: none;
+            box-shadow: none;
+            transition: none;
+            visibility: hidden;
+        }
+
+        @include e(item) {
+            @extend %item;
+        }
+
+        @include e(item, active) {
+            @extend %item;
```

Comment:
When you refactor the item to be its own block, make sure that the `.igx-nav-drawer-item` class is always set to the item, regardless of whether it's active or not. Then you shouldn't need to extend the %item placeholder class when the item is active:

```scss
@include b(igx-nav-drawer-item) {
  @include m(active) {
    @extend %active;
  }
}
```